pkg/build/api/v1/types.go

```diff
@@ -303,6 +303,9 @@ type BuildStrategy struct {
 
 	// CustomStrategy holds the parameters to the Custom build strategy
 	CustomStrategy *CustomBuildStrategy `json:"customStrategy,omitempty" description:"holds parameters to the Custom build strategy"`
+
+	// ExternalStrategy holds the parameters to the External build strategy
+	ExternalStrategy *ExternalBuildStrategy `json:"externalStrategy,omitempty" description:"holds parameters to the External build strategy"`
```

Comment:
no.  the build steps are defined in the jenkinsfile (or whatever mechanism the pipeline engine of choice uses to define a workflow).  This buildconfig is purely a wrapper to that external workflow/pipeline process.  it doesn't care about individual steps.

some of the pipeline steps might be represented by other "normal" buildconfigs (eg an s2 build) but that's not necessarily always going to be the case.